docs/developer/README.md

```diff
@@ -1,5 +1,5 @@
 # Requirements
-* Go >= 1.17
+* Go >= 1.18
 * kubectl >= 1.17
```

Comment:
These other requirements are wrong since we just updated our tools. We don't know if things work with kubectl 1.17 (well, they won't) since we're on 1.24. And same for kustomize 4.0.5, which I don't think works.

Reply:
Right, I didn't pay attention to this (I have more recent versions installed anyway).

I will set:
- kubectl to 1.23 (based on the [version skew policy docs](https://kubernetes.io/releases/version-skew-policy/#kubectl))
- Kustomize to 4.5.7. Maybe older versions work, but I don't think trying them is worth the time. Also it's the current version installed by package managers (brew, etc.)
- Kind to 0.15.0, for the same reasons.

I'll also update `default_kind_node_version` in `setup-kind-multicluster.sh`.